src/execution/execute-afm.convert.ts

```diff
@@ -0,0 +1,75 @@
+// (C) 2007-2019 GoodData Corporation
+
+import { AFM, ExecuteAFM } from '@gooddata/typings';
+import { unset, cloneDeep } from 'lodash';
+
+function convertElementsArray(arr: string[], isText?: boolean): ExecuteAFM.AttributeElements {
+    return isText ? { values: arr } : { uris: arr };
+}
+
+function convertFilter(filter: AFM.CompatibilityFilter): ExecuteAFM.CompatibilityFilter {
+    if (AFM.isPositiveAttributeFilter(filter)) {
+        const clonedFilter = cloneDeep(filter) as ExecuteAFM.IPositiveAttributeFilter;
```

Comment:
Usually using "as" means that somethings not quite right... which looks to me like this case as it lies about object type. Cloned object is not `ExecuteAFM.IPositiveAttributeFilter` as it still contains 'textFilter' attribute... 

also using unset(newFilter, 'textFilter'); is not type-safe 

also this solution could pass secretly new attributes from `AFM.CompatibilityFilter` to `ExecuteAFM.CompatibilityFilter` if somebody forget to "unset" them...

please consider as a solution this: 

```
        const { positiveAttributeFilter: oldFilter } = filter;

        return {
            positiveAttributeFilter: {
                displayForm: oldFilter.displayForm,
                in: convertElementsArray(oldFilter.in, oldFilter.textFilter)
            }
        };
```

Reply:
while i'm not insulted with a cast used as a tech detail inside a pure function I get your point :) 

+ the suggested approach looks better in the code anyway